system/ee/ExpressionEngine/Library/CP/FileManager/Traits/FileManagerTrait.php

```diff
@@ -293,22 +293,38 @@ function ($key) {
         $data = array();
         $missing_files = false;
 
-        $destinationsToEagerLoad = [];
+        // Group files by destination and eager load first
+        $eagerLoads = array_reduce($files->asArray(), function ($carry, $file) use ($member) {
+            if ($file->UploadDestination->adapter != 'local' && $file->UploadDestination->exists()) {
+                if(!array_key_exists($file->upload_location_id, $carry)) {
```

Comment:
Would it be better to swap these lines? I think most of the times the files would be sharing upload destination, so checking whether we already have the info before going into UploadDestination details should be more effective. Also might be worth checking if UploadDestination is not null - for edge cases when the folder record has been removed, but the file record stays - just so it wouldn't bomp out

Reply:
I think this is fine the way it is, that line 298 is just the old line 306 (https://github.com/ExpressionEngine/ExpressionEngine/pull/3630/files/6135f94929ff29b3d01e8a8b44d200d8a65e8b75#diff-cd5b889f5e6fcff6954cdafc0347f4ecdc532ad6e03ce5c5d115ce956fc32644L306) and the UploadDestination relationship gets eager loaded so the performance impact should be trivial.  We would also have to do this same check later when $carry hasn't been set yet and I think this way is a little more readable.